### ISSUE TITLE
[HOLD for payment 2023-08-03] [$1000] NEW FEATURE: Don't truncate long room names in room details page

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#22613]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action performed:

1. Click Global create > new room, and enter a long room name (more than 50 characters).
2. Notice that the full room name is listed in the welcome message, and truncated in the room header.
3. Click into room details and notice that the room name in room details is also truncated.

## Expected result:
We'd show the [full room name in room details](https://expensify.slack.com/archives/C01GTK53T8Q/p1689002306811229?thread_ts=1686863510.616919&cid=C01GTK53T8Q), as this should be the source of truth for what the room actually is. If the room name is long, we should just break it into multiple lines in the room details.
## Actual result:
We truncate the room name in room details. Meaning you cannot see the full room name there.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
![image](https://github.com/Expensify/App/assets/43996225/b9771900-cd04-426b-9d2e-ef276129741b)
![image - 2023-07-10T170007 031 (1)](https://github.com/Expensify/App/assets/43996225/2c89bcdd-418c-4592-a312-4066921707c7)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @joekaufmanexpensify 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689026712373409

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f72d5afba7910b13</li>
        <li>Upwork Job ID: 1678728351340113920</li>
        <li>Last Price Increase: 2023-07-11</li>
    </ul>
</details>

**Comment by jliexpensify** [comment]
@shawnborton let me know if you'd like this as External!

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Don't truncate the room name in the details page which help user to see full room name

### What is the root cause of that problem?
Currently, we are passing numberOfLines to 1 for showing the title which truncates the text and only shows a single line.
https://github.com/Expensify/App/blob/38890e2e3c32b52dad11a7dfc09494924519946b/src/pages/ReportDetailsPage.js#L153

### What changes do you think we should make in order to solve the problem?

1. We only need to set the numberOfLines to 1 when it is not a chat room. Setting to 0 will result in unsetting this value, which means that no lines restriction will be applied.
`numberOfLines={isChatRoom ? 0 : 1}`
https://github.com/Expensify/App/blob/38890e2e3c32b52dad11a7dfc09494924519946b/src/pages/ReportDetailsPage.js#L153

2. To allow it to break in multiple lines we also need to take out the style `styles.pre` from here.
https://github.com/Expensify/App/blob/38890e2e3c32b52dad11a7dfc09494924519946b/src/pages/ReportDetailsPage.js#L154
`isChatRoom ? undefined : styles.pre`

**Note**: It is applied to all platforms, if requires only for small devices then we need to add `withWindowDimensions` and with `isChatRoom` condition in above 2 points we also need to add `&& props.isSmallScreenWidth`.

<details>
<summary>Result</summary>

![Simulator Screen Shot - iPhone 14 - 2023-07-11 at 15 30 22](https://github.com/Expensify/App/assets/14358475/8bc63cb3-1253-4581-9a8e-b126573d0a02)

</details>

**Comment by shawnborton** [comment]
Yup we can do this external, thanks!

**Comment by mananjadhav** [comment]
Thanks for the proposals folks. For consistency, I think we should focus on solving for all the platforms and not just native apps. 

Both the proposals don't fully solve for all the platforms. @dhairyasenjaliya @Pujan92 Can you update your proposals after testing once?  Looking forward to it. Also I would recommend sending out a fresh comment, so that we don't end up in a conflict of which proposal came first. Thank you for your patience on this.

**Comment by Pujan92** [comment]
@mananjadhav my proposal is actually a generic one that behaves the same for all platforms.

**Comment by mananjadhav** [comment]
@Pujan92 still truncated on web using your proposal.

<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/3069065/d9499647-d0da-40fb-ae09-5c23915656df">


```diff
--- a/src/pages/ReportDetailsPage.js
+++ b/src/pages/ReportDetailsPage.js
@@ -150,8 +150,8 @@ function ReportDetailsPage(props) {
                                     fullTitle={ReportUtils.getReportName(props.report)}
                                     displayNamesWithTooltips={displayNamesWithTooltips}
                                     tooltipEnabled
-                                    numberOfLines={1}
-                                    textStyles={[styles.textHeadline, styles.textAlignCenter, styles.pre]}
+                                    numberOfLines={isChatRoom ? 0 : 1}
+                                    textStyles={[styles.textHeadline, styles.textAlignCenter, isChatRoom ? undefined : styles.pre]}
                                     shouldUseFullTitle={isChatRoom || isPolicyExpenseChat || isThread}
                                 />
                             </View>
```

**Comment by mananjadhav** [comment]
But the display name component is component for room as well as chats, won't that cause an issue?

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Don't truncate the room name in the details page which help user to see full room name

### What is the root cause of that problem?
Currently, we are passing numberOfLines to 1 for showing the title which truncates the text and only shows a single line.
https://github.com/Expensify/App/blob/38890e2e3c32b52dad11a7dfc09494924519946b/src/pages/ReportDetailsPage.js#L153

### What changes do you think we should make in order to solve the problem?

1. We only need to set the numberOfLines to 1 when it is not a chat room. Setting to 0 will result in unsetting this value, which means that no lines restriction will be applied.
`numberOfLines={isChatRoom ? 0 : 1}`
https://github.com/Expensify/App/blob/38890e2e3c32b52dad11a7dfc09494924519946b/src/pages/ReportDetailsPage.js#L153

2. To allow it to break in multiple lines we also need to take out the style `styles.pre` from here.
https://github.com/Expensify/App/blob/38890e2e3c32b52dad11a7dfc09494924519946b/src/pages/ReportDetailsPage.js#L154
`isChatRoom ? undefined : styles.pre`

3. Update `numberOfLines=1` with `numberOfLines=this.props.numberOfLines` in the `DisplayNames` to take the dynamic value from parent.
https://github.com/Expensify/App/blob/f2d38d4c42b5e572ad460f53850fb47b67ffc9ef/src/components/DisplayNames/index.js#L85

<details>
<summary>Result</summary>

![Simulator Screen Shot - iPhone 14 - 2023-07-11 at 15 30 22](https://github.com/Expensify/App/assets/14358475/8bc63cb3-1253-4581-9a8e-b126573d0a02)

<img width="1438" alt="Screenshot 2023-07-11 at 10 59 33 PM" src="https://github.com/Expensify/App/assets/14358475/7315984c-726f-4d5a-8ea8-a7f6473a7fb6">

</details>

**Comment by Pujan92** [comment]
> But the display name component is component for room as well as chats, won't that cause an issue?

Only for Room we are taking out the whiteSpace: pre style so I think it won't create any issue.

**Comment by mananjadhav** [comment]
Thanks for the comments folks. Yeah I saw that on @Pujan92's proposal but I missed reading that in yours @dhairyasenjaliya.

Considering both the proposals, I think @Pujan92's [proposal](https://github.com/Expensify/App/issues/22613#issuecomment-1631226608) is better to unset the `numberOfLines`, instead of hard-coding it.

I am a bit concerned about the comment, `// Tokenization of string only support 1 numberOfLines on Web`, so we'll have to test that we don't break something and trace it back to the original PR when this comment was added.


🎀 👀 🎀  C+ Reviewed.

**Comment by mananjadhav** [comment]
@dhairyasenjaliya Your proposal highlights, `We need to define max number`, where as the other proposal mentions unsetting the value. Obviously for smaller PRs the properties, etc. could be same and I wouldn't call it _same proposals_. I hope this clarifies.

**Comment by jliexpensify** [comment]
Ah, Chirag is on leave now - I shall reassign another Engineer!

**Comment by jliexpensify** [comment]
Hi @Julesssss - just wondering if you could take over for Chirag. We have a proposal [here](https://github.com/Expensify/App/issues/22613#issuecomment-1631242188) - cheers!

**Comment by mananjadhav** [comment]
@Julesssss quick bump on this one.

**Comment by Julesssss** [comment]
Oop,s missed that last comment. Will review tomorrow

**Comment by jliexpensify** [comment]
Adding Chirag back so he gets the credit when this is closed.

**Comment by Pujan92** [comment]
@mananjadhav shall we hold this issue until this [PR](https://github.com/Expensify/App/pull/18507) gets merged?

Reason - for android resetting the numberOfLines={0} gives an error and the above PR seems to have a patch for numberOfLines. I tested that branch code and seems it is working in Android with it.

![Screenshot_1689934970](https://github.com/Expensify/App/assets/14358475/df85685f-c611-406d-ab84-d324cc82eec5)

**Comment by mananjadhav** [comment]
I agree let's put this on hold.

**Comment by mananjadhav** [comment]
@Pujan92 I saw the PR,it allows number | undefined. how about we set this to undefined? instead of 0. Does this still break it?

**Comment by Pujan92** [comment]
@mananjadhav I haven't tried undefined yet but as per the react doc it's preferred to pass default value 0.
https://reactnative.dev/docs/text#numberoflines

**Comment by mananjadhav** [comment]
We just tried in this issue, and it seems to work. https://github.com/Expensify/App/pull/23246#issuecomment-1645573411

Any thoughts @Julesssss ?

**Comment by Julesssss** [comment]
My only concern with holding is that the RN bumps sometimes take quite a while.

**Comment by mananjadhav** [comment]
yeah and we used `this.props.numberOfLines || undefined`, and it worked fine.

**Comment by Julesssss** [comment]
Lets just do [this](https://github.com/Expensify/App/issues/22613#issuecomment-1645655627) then? 

I don't think it's a huge deal either way

**Comment by mananjadhav** [comment]
Agreed. @Pujan92 ^ ?

**Comment by Pujan92** [comment]
Yes, will raise a PR soon!

**Comment by Pujan92** [comment]
@mananjadhav seems undefined also giving me an error.

**Comment by mananjadhav** [comment]
@Pujan92 Please check this [PR](https://github.com/Expensify/App/pull/23246/files) for reference.

**Comment by b4s36t4** [comment]
I'm just wondering passing `0` as numberOfLines does make any sense? isn't it weird to understand we're saying `0` lines and it will show all the lines of text? why not just pass `undefined` or `null` this makes somewhat sense in my view (came here after seeing PR)?

**Comment by mananjadhav** [comment]
This is more like a feature request. Hence we don't have any tagged PR. Because we shouldn't end up working on adding the limit again, I think we should add a regression test.

**Regression Test:**

1. Open FAB, and click on `New Room`
2. Create a room with a very long name (> 100 characters)
3. Open the room chat
4. Click on room name in the header to open report details
5. Verify that room name won't truncate and shows entire room name text

@Julesssss @jliexpensify wdyt? 

Also this is eligible for the timeline bonus, because the change was done well within the timeline, but we got blocked by another PR conflict.

**Comment by jliexpensify** [comment]
Will let @Julesssss weigh in with the final say on timeline bonus - I'll summarise payments once he weighs in!

**Comment by Julesssss** [comment]
Regular payment here @jliexpensify

**Comment by jliexpensify** [comment]
Ok @Julesssss is this correct?

Payments:

- Contributor: @Pujan92 $1000
- C+: @mananjadhav $1000

Upworks job [here](https://www.upwork.com/jobs/~01f72d5afba7910b13)

**Comment by mananjadhav** [comment]
@jliexpensify  It should be $1500.

**Comment by Julesssss** [comment]
> @jliexpensify It should be $1500, but let's wait for @Julesssss to ack.

Why is that?

**Comment by mananjadhav** [comment]
@Julesssss I raised this comment [here](https://github.com/Expensify/App/pull/23357#issuecomment-1646847028) too during the PR. We had the PR ready but had conflicts. The conflict was primarily a regression, which got this issue delayed. I saw your 👍 in the comment as well as  I highlighted the same [here too.](https://github.com/Expensify/App/issues/22613#issuecomment-1656852705)

**Comment by Julesssss** [comment]
Ah yeah, I remember now. 👍

**Comment by mananjadhav** [comment]
Thanks. Can you comment and confirm it to @jliexpensify so that he can put up a summary on the issue?

**Comment by Julesssss** [comment]
@jliexpensify I hereby declare this is worthy of the bonus 😄

**Comment by jliexpensify** [comment]
[Edited](https://github.com/Expensify/App/issues/22613#issuecomment-1657865570)

**Comment by mananjadhav** [comment]
Raised my request on the NewDot.

**Comment by jliexpensify** [comment]
Pujan paiud, job closed. Working on Regression GH now.

**Comment by JmillsExpensify** [comment]
Reviewed details for @mananjadhav. These details are accurate based on summary from Business Reviewer and are now approved for payment in NewDot.
